File: src/main/kotlin/gbl/tag/type/certificate/GblCertificateEcdsaP256.kt
package gbl.tag.type.certificate

import parser.data.tag.GblType
import gbl.tag.TagHeader
import gbl.tag.Tag
import gbl.tag.TagWithHeader

data class GblCertificateEcdsaP256(
    override val tagHeader: TagHeader,
    override val tagType: GblType,
    override val tagData: ByteArray,
    val certificate: ApplicationCertificate
): Tag, TagWithHeader {
    override fun copy(): Tag {
        return GblCertificateEcdsaP256(
            tagHeader = tagHeader,
            tagType = tagType,
            tagData = tagData,
            certificate = certificate
        )
    }
}

File: python-library/tag/type/certificate/gbl_certificate_ecdsa_p256.py

"""
GBL Certificate ECDSA P256 tag implementation
Exact conversion from Kotlin GblCertificateEcdsaP256.kt
"""

from dataclasses import dataclass
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from ...gbl_type import GblType
    from ...tag_header import TagHeader
    from ...tag import Tag

from .application_certificate import ApplicationCertificate


# Імпорти з інших модулів (будуть додані пізніше):
# from ...tag import Tag
# from ...tag_with_header import TagWithHeader


@dataclass
class GblCertificateEcdsaP256:
    """GBL Certificate ECDSA P256 tag"""

    tag_header: 'TagHeader'
    tag_type: 'GblType'
    tag_data: bytes
    certificate: ApplicationCertificate

    def copy(self) -> 'Tag':
        """Create a copy of the tag"""
        return GblCertificateEcdsaP256(
            tag_header=self.tag_header,
            tag_type=self.tag_type,
            tag_data=self.tag_data,
            certificate=self.certificate
        )